pulsar-client/src/main/java/org/apache/pulsar/client/impl/PulsarClientImpl.java

```diff
@@ -1012,7 +1013,8 @@ public MemoryLimitController getMemoryLimitController() {
                 @SuppressWarnings("rawtypes") Schema finalSchema = schema;
                 return schemaInfoProvider.getLatestSchema().thenCompose(schemaInfo -> {
                     if (null == schemaInfo) {
-                        if (!(finalSchema instanceof AutoConsumeSchema)) {
+                        if (!(finalSchema instanceof AutoConsumeSchema)
+                            && !(finalSchema instanceof KeyValueSchema)) {
```

Comment:
one of keySchema and valueSchema is `AutoConsumeSchema` need this right?

Reply:
yes, this comes from the fact that above the schema returned `true` to `requireFetchingSchemaInfo`and so it is a dynamic schema

https://github.com/apache/pulsar/blob/0649ef80c5f07fc4a81f909c5c4c79dbef641ead/pulsar-client/src/main/java/org/apache/pulsar/client/impl/PulsarClientImpl.java#L1012